### PR TITLE
[AP-372] Modify public healthcheck to use /health endpoint by default and support optional checks

### PR DESCRIPTION
**Pull request by kibertoad**
## Changes

Align with https://github.com/lokalise/platform-handbook/blob/main/processes/health-check-requirements.md

## Checklist

- [X] Apply one of following labels; `major`, `minor`, `patch` or `skip-release`
- [X] I've updated the documentation, or no changes were necessary
- [X] I've updated the tests, or no changes were necessary

### CODE REVIEW

File: lib/plugins/publicHealthcheckPlugin.ts [hunk 1]
```diff
@@ -4,42 +4,61 @@ import fp from 'fastify-plugin'
...
         }
+
+        if (isHealthy) {
+          if (
+            results.find((entry, index) => {
+              return entry.error && !opts.healthChecks[index].isMandatory
+            })
+          ) {
+            isHealthy = false
+            isPartiallyHealthy = true
```
Comment (lokankr) [inline lib/plugins/publicHealthcheckPlugin.ts:54]: Although tests looks correct, this logic is a bit tricky to follow. Only if `isHealthy === true` then `isPartiallyHealthy` can become `true`? 🤔

Could this be done in a single loop, and then
- If any healthcheck passes then set `isPartiallyHealthy` to `true`.
- If any healthcheck fails then set `isHealthy` to `false`
?

**Review by CarlosGamero — APPROVED** [review]

**Review by lokankr — APPROVED** [review]

File: lib/plugins/publicHealthcheckPlugin.ts (hunk 1 above)
Comment (rupert-mckay) [reply lib/plugins/publicHealthcheckPlugin.ts]: Agreed that this is very hard to follow. Here is a rewrite:

```ts
    handler: async (_, reply) => {
      const mandatoryChecks = opts.healthChecks.filter(({ isMandatory }) => isMandatory)
      const optionalChecks = opts.healthChecks.filter(({ isMandatory }) => !isMandatory)

      const [
        mandatoryCheckResults,
        optionalCheckResults
      ] = await Promise.all([
        Promise.all(mandatoryChecks.map((healthCheck) => healthCheck.checker(app))),
        Promise.all(optionalChecks.map((healthCheck) => healthCheck.checker(app))),
      ])

      // If any mandatory check has an error, then the health check has failed
      if(mandatoryCheckResults.some((result) => !!result.error)) {
        return reply.status(500).send({
          ...responsePayload,
          heartbeat: 'FAIL'
        })
      }

      // If any optional check has an error, then we are partially healthy
      if(optionalCheckResults.some((result) => !!result.error)) {
        return reply.status(200).send({
          ...responsePayload,
          heartbeat: 'PARTIALL_HEALTHY'
        })
      }

      // No checks had an error. Note this includes the case that there were no checks at all.
      return reply.status(200).send({
        ...responsePayload,
        heartbeat: 'HEALTHY'
      })
    },

```
